cmd/goplugin-aws/resource/iam_role.go

```diff
@@ -0,0 +1,150 @@
+package resource
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/hashicorp/go-hclog"
+)
+
+//IamRole is the managed resource
+type IamRole struct {
+	Description              *string
+	RoleName                 string
+	AssumeRolePolicyDocument string
+	Path                     *string
+	Tags                     map[string]string
+}
+
+//RoleHandler creates, reads and deletes the IamRole Resource
+type RoleHandler struct {
+}
+
+// Create an iam.Role
+func (h *RoleHandler) Create(desired *IamRole) (*IamRole, string, error) {
+	log := hclog.Default()
+	if log.IsDebug() {
```

Comment:
This isn't really a comment about this PR, but this debug check feels like something we should not have to do.